test/api/core/entrypoints/clientConnection.test.js

```diff
@@ -1,20 +1,37 @@
 const
   should = require('should'),
-  ClientConnection = require('../../../../lib/api/core/entrypoints/clientConnection');
+  ClientConnection = require('../../../../../lib/api/core/entrypoints/embedded/clientConnection');
 
 describe('core/clientConnection', () => {
   describe('#constructor', () => {
+    let
+      headers,
+      connection;
+
+    beforeEach(() => {
+      headers = { foo: 'bar' };
+      connection =
+        new ClientConnection('protocol', ['ip1', 'ip2'], headers);
+    });
+
     it('should throw if ips is not an array', () => {
       return should(() => new ClientConnection('protocol', 'ips'))
         .throw(TypeError, {message: 'Expected ips to be an Array, got string'});
     });
 
     it('should set headers', () => {
-      const
-        headers = {foo: 'bar'},
-        connection = new ClientConnection('protocol', ['ip1', 'ip2'], headers);
-
       should(connection.headers).be.exactly(headers);
     });
+
+    it('should be frozen', () => {
```

Comment:
(nitpicking) you can also test using [Object.isFrozen](https://devdocs.io/javascript/global_objects/object/isfrozen)